projects/route-analyzer/src/lib/route-analyzer.ts

```diff
@@ -179,15 +182,58 @@ function routeCallToRoutes(moduleRouteCall: ModuleRouteCall, typeChecker: ts.Typ
                 route.loadChildren = getModulefromLoadChildrenCall(prop.initializer);
             } else if (propName === ROUTE_PROP.COMPONENT) {
                 // Maybe not really needed, but just in case we add the component name
+                // What is more useful is the selector for the component
+                // Analyze the imports, looking for the component name so we can find its file
+                // And get its tag name
                 route.component = prop.initializer.getText();
+                const tagName = getTagName(prop.initializer, typeChecker);
+                if (tagName) {
+                    // Don't lose existing data that may or may not have been added, depending on the order
+                    // that properties were declared
+                    route.data = { ...(route.data || {}), tagName };
```

Comment:
Isn't it better to  add a function that preserves the old data and adds the new one?
The function name will suggest that we don't lose the existing data